app.js

```diff
@@ -40,4 +42,57 @@ function timeBetweenDates(a = '25 May 2023', b = Date.now(), units = 'days') {
 console.log(timeBetweenDates('02 Aug 1985', '03 Aug 1985', 'seconds'));
 console.log(timeBetweenDates('31 Jan 2022', '03 Feb 2021', 'days'));
 console.log(timeBetweenDates('31 Jan 2022', '03 Feb 2021', 'years'));
-console.log('I study JavaScript for ' + (timeBetweenDates() + 1) + ' days');
\ No newline at end of file
+console.log('I study JavaScript for ' + (timeBetweenDates() + 1) + ' days');
+
+
+// Task2
+
+const priceData = {
+    Apples: '23.4',
+    BANANAS: '48',
+    oRAnGEs: '48.7584', };
+
+function optimizer(data) {
+    let entriesArray = Object.entries(data);
+    let keysToLowerCaseArray = entriesArray.map(arr => arr[0].toLowerCase());
+    let valuesToNumber = entriesArray.map(arr => parseFloat(arr[1]).toFixed(2));
+    
+    let finalArray = [];
+    for (let i = 0; i < keysToLowerCaseArray.length; i++) {
+        let combinedArray = [];
+        combinedArray.push(keysToLowerCaseArray[i]);
+        combinedArray.push(valuesToNumber[i]);
+        finalArray.push(combinedArray);
+    }
+    let optimizedObject = Object.fromEntries(finalArray);
+
+    return optimizedObject;
+}
+
+console.log(optimizer(priceData));
+
+
+// Task3
+
+function iterativeOddSumTo(n) {
+    let sum = 0;
+    for (let i = 0; i <= n; i++) {
+        if  (i % 2 !== 0) {
+            sum += i; 
+        }               
+    }
+    return sum;
+};
+
+function recursiveOddSumTo(n) {
+    if (n <= 0) {
+        return 0; 
+    } else if (n % 2 === 0) {
+        return recursiveOddSumTo(n - 1);
+    } else {
+        return n + recursiveOddSumTo(n - 2); 
+    }
+}
```

Comment:
Не має потреби в if...else
```suggestion
function recursiveOddSumTo(n) {
    if (n <= 0) {
        return 0;
    }
    if (n % 2 === 0) {
        return recursiveOddSumTo(n - 1);
    }
    return n + recursiveOddSumTo(n - 2);
}
```

Можна спростити запис